backend/infrahub/core/attribute.py

```diff
@@ -255,6 +256,14 @@ def validate_content(cls, value: Any, name: str, schema: AttributeSchema) -> Non
             if len(value) > max_length:
                 raise ValidationError({name: f"{value} must have a maximum length of {max_length!r}"})
 
+        if isinstance(schema.parameters, NumberAttributeParameters):
+            min_value = schema.parameters.min_value
+            if min_value is not None and value < min_value:
+                raise ValidationError({name: f"{value} is lower than the minimum allowed value {min_value!r}"})
+            max_value = schema.parameters.max_value
+            if max_value is not None and value > max_value:
+                raise ValidationError({name: f"{value} is higher than the maximum allowed value {max_value!r}"})
+
```

Comment:
Instead of having the logic here I think we could have methods in NumberAttributeParameters and have them raise these errors if needed.